packages/x-data-grid-pro/src/tests/events.DataGridPro.test.tsx

```diff
@@ -344,7 +344,14 @@ describe('<DataGridPro /> - Events params', () => {
         rowCount={50}
       />,
     );
-    act(() => apiRef.current.publishEvent('scrollPositionChange', { left: 0, top: 3 * 52 }));
+    act(() => {
+      apiRef.current.publishEvent('renderedRowsIntervalChange', {
+        firstRowIndex: 3,
+        lastRowIndex: 6,
+        firstColumnIndex: 0,
+        lastColumnIndex: 0,
+      });
+    });
```

Comment:
Why is this test modified?

Reply:
Seemed like a faulty premise for the test, but honestly, but in hindsight, I'm not sure why it previously worked to begin with.

Lazy loader only subscribes to `renderedRowsIntervalChange`:
https://github.com/mui/mui-x/blob/1a079e18fad1ddc2bf4a49f476bb5c68b47de1a2/packages/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts#L134-L138

`scrollPositionChange` shouldn't change actual scroll position nor trigger `renderedRowsIntervalChange`, so how can it even trigger a new slice to be loaded?

Maybe you understand it better, but I tried to test it in the browser and everything worked fine with actual scrolling.

Reply:
This test gave me so much grief trying to trace in the code base how `scrollPositionChange` can possibly trigger lazyLoader. And guess what? It doesn't. 

Ran it on the mater branch and removing the `publishEvent` line will still make the pass. But for whatever reason, it will always try to load all the rows minus the first 3 that it has – tha args to onFetchRows are:
```
Object{firstRowToRender: 3, lastRowToRender: 500, sortModel: [], filterModel: Object{items: [], logicOperator: 'and', quickFilterValues: [], quickFilterLogicOperator: 'and'}}
```

Again, this is on the master branch, for the lack of any confusion.

Reply:
The test is still testing something useful that broke, just not what it says on the label currently. I reduced the amount `updateDimensions` calls, which called `updateRenderContext` ahead of dimensions state being updated, which prevented `lazyLoader` from working directly on mount (scroll still worked).

Fixed it now, and improved the tests – split it into two and improved the checks to more than just checking if fetchRows has been called.